virtual_rainforest/models/abiotic/wind.py

```diff
@@ -0,0 +1,553 @@
+
+    # TODO: find correct fill value
+    """
+
+    return (
+        (friction_velocity / 0.4)
+        * np.log((wind_heights - zero_plane_displacement) / roughness_length_momentum)
+        + diabatic_correction_momentum_above
+    ).fillna(0)
+
+
+def calculate_wind_below_canopy(
+    canopy_node_heights: DataArray,
+    wind_profile_above: DataArray,
+    wind_attenuation_coefficient: DataArray,
+    canopy_height: DataArray,
+) -> DataArray:
+    """Calculate wind profile below canopy.
+
+    Implementation after :cite:t:`maclean_microclimc_2021`.
+    The top of canopy windspeed is taken from the above canopy wind profile, which must
+    be ordered from highest to lowest level. The lowest level is used here as the
+    top-of-canopy wind speed.
+
+    Args:
+        copy_node_heights: heights of canopy nodes, [m]
+        wind_profile_above: wind profile above canopy, [m s-1]
+        wind_attenuation_coefficient: wind attenuation coefficient, dimensionless
+        canopy_height: canopy height, [m]
+
+    Returns:
+        wind profile below canopy, [m s-1]
+
+    # TODO: find correct fill value
+    """
+    top_of_canopy_wind_speed = wind_profile_above[:, -1]
+
+    return (
+        top_of_canopy_wind_speed
+        * np.exp(
+            wind_attenuation_coefficient * ((canopy_node_heights / canopy_height) - 1)
+        )
+    ).fillna(0)
+
+
+def calculate_zero_plane_displacement(
+    canopy_height: DataArray,
+    leaf_area_index: DataArray,
+    zero_plane_scaling_parameter: float = WindConstants.zero_plane_scaling_parameter,
+) -> DataArray:
+    """Calculate zero plane displacement.
+
+    The zero plane displacement height of a vegetated surface is the height at which the
+    wind speed would go to zero if the logarithmic wind profile was maintained from the
+    outer flow all the way down to the surface (that is, in the absence of the
+    vegetation when the value is set to zero). Implementation after
+    :cite:t:`maclean_microclimc_2021`.
+
+    Args:
+        canopy_height: canopy height, [m]
+        leaf_area_index: leaf area index, [m m-1]
+        zero_plane_scaling_parameter: Control parameter for scaling d/h
+            :cite:p:`raupach_simplified_1994`
+
+    Returns:
+        zero place displacement height, [m]
+    """
+    plant_area_index = leaf_area_index.sum(dim="canopy_layers")
+
+    # calculate in presence of vegetation
+    plant_area_index_displacement = plant_area_index.where(plant_area_index > 0)
+    zero_place_displacement = DataArray(
+        (
+            1
+            - (
+                1
+                - np.exp(
+                    -np.sqrt(
+                        zero_plane_scaling_parameter * plant_area_index_displacement
+                    )
+                )
+            )
+            / np.sqrt(zero_plane_scaling_parameter * plant_area_index_displacement)
+        )
+        * canopy_height,
+        dims="cell_id",
+    )
```

Comment:
The maths here are obscured because of `black` breaking down the lines in a pretty insane way. I would suggest to calculate the two `sqrt` separately, assigning them to a (short-named) variable, and then use those in the calculation. Hopefully, the layout will be more readable.